website/src/components/UI/Affix.tsx

```diff
@@ -0,0 +1,99 @@
+import type { CSSProperties, ReactNode, FC } from 'react';
+import React, { useState, useEffect } from 'react';
+import styled from 'styled-components';
+
+import { getDomStyle } from '../../utils';
+
+interface Props {
+  style: CSSProperties;
+  children: ReactNode;
+}
+
+const AffixContent = styled.div`
+`;
+
+const getPositionStyle = (
+  hasCssSticky: boolean,
+  defaultStyles: CSSProperties = {},
+):CSSProperties => {
+  const { width } = defaultStyles;
+  const positionStyle: CSSProperties = hasCssSticky ? {
+    position: 'sticky',
+    marginLeft: `-${width}px`,
+    display: 'inline-block',
+    float: 'left',
+  } : {
+    position: 'absolute',
+  };
+
+  return {
+    ...positionStyle,
+    ...defaultStyles,
+  };
+};
+
+const Affix: FC<Props> = (props) => {
+  const { style, children } = props;
+  const [hasCssSticky, SetHasCssSticky] = useState(true);
+  useEffect(() => {
+    SetHasCssSticky(CSS.supports('position', 'sticky'));
+  });
```

Comment:
You may need the help of this document https://reactjs.org/docs/hooks-reference.html#timing-of-effects.

Since I didn't write most of the original code, and I haven't tried to improve them, many of them are negative examples, so please don't learn how they are written.

Reply:
there is look like escape SSR build error? I met it today..😂, let me check how to modify~ https://app.netlify.com/sites/apache-apisix/deploys/628a20980921c8000875641f

Reply:
You may need to focus on dependency arrays 🤔

Reply:
> You may need to focus on dependency arrays 🤔

It was a good learning experience, THX ☺️ 

https://github.com/streamich/react-use/blob/e53ca94a0b1f20270b0f75dc2ca1fecf1e119dde/src/useEffectOnce.ts#L3